engine/runtime-compiler/src/main/java/org/enso/compiler/phase/IdMapProcessor.java

```diff
@@ -0,0 +1,45 @@
+package org.enso.compiler.phase;
+
+import java.util.Map;
+import java.util.UUID;
+import org.enso.compiler.core.ir.Expression;
+import org.enso.compiler.core.ir.IdentifiedLocation;
+import org.enso.compiler.core.ir.Location;
+import org.enso.compiler.core.ir.Module;
+import org.enso.compiler.data.IdMap;
+import scala.Some$;
+
+public final class IdMapProcessor {
+
+  private IdMapProcessor() {}
+
+  /**
+   * Applies the map with external ids to the module ir.
+   *
+   * @param ir the module ir.
+   * @param idMap the provided IdMap.
+   */
+  public static Module updateIr(Module ir, IdMap idMap) {
+    if (idMap == null) {
+      return null;
```

Comment:
Shouldn't this return `ir` itself and the usage site should check for null `idMap`?

Reply:
The non-null return value indicates that the `IR` was updated we need to re-run the compiler passes on it

Reply:
In most cases it is unchanged, and we can skip the compilation